src/main/java/com/autosale/shop/service/impl/ProductServiceImpl.java

```diff
@@ -0,0 +1,81 @@
+package com.autosale.shop.service.impl;
+
+import com.autosale.shop.model.Product;
+import com.autosale.shop.model.ProductStatus;
+import com.autosale.shop.repository.ProductRepository;
+import com.autosale.shop.service.ProductService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.List;
+
+import static org.springframework.security.core.context.SecurityContextHolder.getContext;
+
+@Service
+@RequiredArgsConstructor
+public class ProductServiceImpl implements ProductService {
+    private final ProductRepository repository;
+
+    @Override
+    public List<Product> findAll() {
+        return repository.findAll();
+    }
+
+    @Override
+    public List<Product> findAllActive() {
+        return repository.findAllActive();
+    }
+
+    @Override
+    public Product findById(int id) {
+        return repository.findById(id).orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, String.format("Product with id %d not found", id)));
+    }
+
+    @Override
+    public List<Product> findAllFromCurrentUser() {
+        return repository.findAllByUserId((Integer) getContext().getAuthentication().getPrincipal());
```

Comment:
Can Principal be converted to int?

Reply:
If it's a question about "Can I simplify Integer to int?" then yes and I'll do it, and if it's a question about "Is principal is int inside?" then again yes, when I create authentication in security filter I use user id, which is integer, as authentication principal